src/endpoints/data_objects/impl.cpp

```diff
@@ -1605,6 +1605,14 @@ namespace
 					const json input{{"logical_path", lpath_iter->second}, {"options", options}};
 
 					auto conn = irods::get_connection(client_info.username);
+
+					if (!fs::client::is_data_object(conn, lpath_iter->second)) {
```

Comment:
Does this prevent touch from creating data objects? If so, should we check for the "no-create" option here?

Reply:
Good catch! I think a better solution is to use `is_collection` instead.

The intent behind that line is to not allow references to collections.

Will change.

Reply:
Decided it was better to test for existence and then check if the target is NOT a data object. The current code is only wrong in the case where the path doesn't exist.

Reply:
Done. Please review and let me know if anything sticks out to you. Thanks.